src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,34 @@
+package duke.command;
+
+import duke.common.CustomException;
+import duke.common.Ui;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.io.IOException;
+
+/**
+ * Adds a deadline task.
+ */
+public class DeadlineCommand extends Command {
+    private static String description;
+    private static String by;
+
+    public DeadlineCommand(String description, String by) {
+        DeadlineCommand.description = description;
+        DeadlineCommand.by = by;
+    }
+    public void execute (TaskList tasks, Ui ui, Storage storage) throws CustomException, IOException {
+        try {
+            Task temp = new Deadline(description, by);
+            tasks.addTask(temp);
+            Storage.appendToFile(temp);
+            Ui.display("Yay! New task added:\n   " + temp +
```

Comment:
Perhaps for cases like these where there are long String of texts, you can consider abstracting it out as private static final String or can consider an ENUM class too! Applies for other parts of ur code too!